### PR TITLE
Fixes #1865: Spawn overflow items into the world

### LINKED ISSUE #1865: Annihilation Plane not correctly spawning overflow items
The Annihilation Plane is not correctly spawning overflowing items into the world as last resort, instead these are currently voided.

### PR DESCRIPTION
**Pull request by yueh**
Fixes #1865

### CODE REVIEW

**Comment by yueh** [comment]
Updated to spawn the entity in the center of the broken block and without any motion.

So it no longers spawns on the border of block and thus gets moved somewhere else as this space is occupied and no motion keeps it inside the same blockspace.

File: src/main/java/appeng/parts/automation/PartAnnihilationPlane.java [hunk 1]
```diff
@@ -359,7 +359,36 @@ private boolean handleOverflow( EntityItem entityItem, IAEItemStack overflow )
...
+	 * @param y y coordinate
+	 * @param z coordinate
+	 * @param overflow the item to spawn
+	 */
+	private void spawnOverflow( IAEItemStack overflow )
+	{
+		if( overflow == null )
+		{
+			return;
+		}
```
Comment (thatsIch) [inline src/main/java/appeng/parts/automation/PartAnnihilationPlane.java]: not sure if this method is the place where to silently reject no overflow, should probably be checked in the loop

Comment (yueh) [inline src/main/java/appeng/parts/automation/PartAnnihilationPlane.java]: Probably fine for this case, I usually just prefer handling this stuff at least also on the lowest level and not have to wrap every call of `PartAnnihilationPlane.spawnOverflow` with a null check. Having one single null check everything needs to go through simply is a bit less code.

I could change it to a `PreCondition` or `assert`. But that then throw an exception under some rare conditions. Also I cannot use assertions in all cases. Like enabled assertions will cause Extra Utilities to not start indev, which I needed to test it this.

**Comment by thatsIch** [comment]
tested on server

**Pull request merged**